mozetl/search/aggregates.py

```diff
@@ -222,15 +222,20 @@ def _drop_source_columns(base):
         return derived
 
     def _select_counts(main_summary, col_name, count_udf=None):
-        derived = main_summary.withColumn(
-            "single_search_count", explode(col("search_counts"))
-        ).filter("single_search_count.count < %s" % MAX_CLIENT_SEARCH_COUNT)
+        ad_fields = ("ad_click", "search_with_ads")
+        if col_name in ad_fields:
+            # special case for ad click probes
+            derived = main_summary
+        else:
+            derived = main_summary.withColumn(
+                "single_search_count", explode(col("search_counts"))
+            ).filter("single_search_count.count < %s" % MAX_CLIENT_SEARCH_COUNT)
```

Comment:
I don't think we really need to specify the ad click fields. Really it's the single search count case which is "special". I think I would prefer something like this:

```py
if col_name == "single_search_count":
    derived = main_summary.withColumn(
                "single_search_count", explode(col("search_counts"))
            ).filter("single_search_count.count < %s" % MAX_CLIENT_SEARCH_COUNT)
else:
    derived = main_summary
```

Reply:
You're right, this is cleaner!